packages/ref-metrics-example/src/ref_metrics_example/example.py

```diff
@@ -0,0 +1,115 @@
+from pathlib import Path
+from typing import Any
+
+import xarray as xr
+from ref_core.metrics import Configuration, MetricResult, TriggerInfo
+
+
+def calculate_annual_mean_timeseries(dataset: Path) -> xr.Dataset:
+    """
+    Calculate the annual mean timeseries for a dataset.
+
+    While this function is implemented here,
+    in most cases the metric calculation will be in the underlying benchmarking package.
+    How the metric is calculated is up to the provider.
+
+    Parameters
+    ----------
+    dataset
+        A path to a CMIP6 dataset.
+
+        This dataset may consist of multiple data files.
+
+    Returns
+    -------
+    :
+        The annual mean timeseries of the dataset
+    """
+    input_files = dataset.glob("*.nc")
+
+    xr_ds = xr.open_mfdataset(list(input_files), combine="by_coords", chunks=None, use_cftime=True)
+
+    annual_mean = xr_ds.resample(time="YS").mean()
+    return annual_mean.mean(dim=["lat", "lon"], keep_attrs=True)
+
+
+def format_cmec_output_bundle(dataset: xr.Dataset) -> dict[str, Any]:
```

Comment:
(Maybe I'm jumping ahead and you're planning on doing this already but not yet while you're prototyping, if so just ignore me). Is it worth creating a class for the output rather than using the ambiguous `dict[str, Any]`? We can always serialise to a dict if we need, but having a class could help communicate what is actually needed.

Reply:
Just saw this: https://github.com/CMIP-REF/cmip-ref/pull/5/files#diff-2f33f70c0be4a6d039d01398ce350b06f838a71a347972a772299bfab43842e7R58

Ignore me, nice stuff

Reply:
I would have resolved the thread already, but for some reason I cannot 🙃